### ISSUE TITLE
Is `clasp` still being actively maintained/developed?

### ISSUE THREAD

**Issue by imthenachoman** [google/clasp#922]
If not, is there anything we can do to help keep it going?

**Comment by grant** [comment]
Hey Nacho Man! Thanks for using `clasp`. :)

I created `clasp` and worked on it awhile ago. _(not working at the big G anymore)_. Fun times.

I'll give my (external) perspective~

---

I wouldn't say that this project is _actively_ maintained, but it's definitely not archived. The API is still up and running, important PRs for this repo get merged and then the CLI gets auto-published to npm.

If there's something you'd like improved with `clasp`, I suggest creating a simple, high-quality PR that fixes a filed issue. I did so just a month ago and got a speedy review and merge. https://github.com/google/clasp/pull/936

**Comment by imthenachoman** [comment]
I see. Thank you.

I have [submitted PRs](https://github.com/google/clasp/pull/885) (#703 and #884) before, value-add ones IMHO, but they just sit there. That's why I thought maybe it was not being maintained.

**Comment by MichaelRWolf** [comment]
With GitHub's recent move beyond beta and to free accounts for Codespaces (and Copilot), I expect that clasp(1) will be getting a  lot more attention.  Developing in the browser-based editor is a non-starter for non-trivial coding.  It took me an hour of searching last night to find clasp(1).  It's been a life-saver.

Perhaps GitHub could do some work to get serious developers onboard GitHub Codespaces, because serious developers don't use non-serious tools.  Codespaces is a serious tool, but the editor is not!

**Comment by imthenachoman** [comment]
Agreed. GAS is the best productivity platform out there -- bar none. I don't think most folks realize how powerful it is. I've been broadcasting it at work and usage has increased exponentially. It solves so many, so, so many business problems. 

And with greater usage, `clasp` is a must. I wouldn't even mind paying for `clasp`, or GAS for that matter.

**Comment by OctanMan** [comment]
As a software developer by-day, discovering GAS and `clasp` has made me feel at home in my attempts to automate some Workspace processes for a non-profit I volunteer at. I only wish its capabilities were more visible - I discovered GAM first which I guess is more in-tune with the sys-admin crowd.

I too am interested to know how much priority Google is giving GAS and `clasp`. Trying to use the `directory_v1` API to set user recovery emails today, I noticed the TS definitions were out of date and didn't include the [`recoveryEmail` property](https://developers.google.com/admin-sdk/directory/reference/rest/v1/users?authuser=0#resource:-user) (despite the npm package being updated just 2 months ago) as it's not clear who has responsibility for that now... 

The Google API Discovery Service (I'm pretty sure) didn't exist upon `clasp`s inception, although it now seems like the most low-effort way to keep the TS defs up-to-date using a tool like this: https://github.com/callmehiphop/discovery-tsd

**Comment by sqrrrl** [comment]
Still alive :) Haven't had a ton of time to work on it, but just pushed a release last week and working on 3.x where we can introduce some breaking changes -- mostly dropping typescript support as it's better to do this as a separate build step before pushing files with clasp.
